insights/client/crypto.py

```diff
@@ -0,0 +1,122 @@
+import subprocess
+
+from .constants import InsightsConstants as constants
+
+logger = logging.getLogger(__file__)
+
+
+def _call_gpg(home, command):
+    """
+    Run /usr/bin/gpg.
+
+    Parameters:
+        home (str): Path to GPG's home directory.
+        command (str): Actual arguments passed to the GPG binary.
+
+    Returns (str, str, int):
+        Standard output, standard error and the return code.
+    """
+    command = (
+        "/usr/bin/gpg --homedir {home} {command}"
+    ).format(home=home, command=command)
+    process = subprocess.Popen(shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    raw_stdout, raw_stderr = process.communicate()
+    stdout, stderr = raw_stdout.decode("utf-8"), raw_stderr.decode("utf-8")
+
+    logger.debug(command)
+    logger.debug("status code {code}".format(code=process.returncode))
+
+    if process.returncode > 0 and stdout:
+        logger.debug("stdout is not empty:\n%s", stdout.strip())
+    if process.returncode > 0 and stderr:
+        logger.debug("stderr is not empty:\n%s", stderr.strip())
+
+    return stdout, stderr, process.returncode
+
+
+def verify_gpg_signed_file(file, signature):
+    """
+    Verify a file that was signed using GPG.
+
+    Parameters:
+        file (str): A path to the signed file.
+        signature (str): A path to the detached signature.
+
+    Returns (dict):
+        ok (str): Whether the signature holds.
+        stdout (str): Standard output of GPG.
+        stderr (str): Standard error of GPG.
+        code (int): Return code of GPG.
+    """
+    if not os.path.isfile(file):
+        message = "cannot verify signature of '{file}', file does not exist".format(file=file)
+        logger.debug(message)
+        return {
+            'gpg': False,
+            'stdout': '',
+            'stderr': '',
+            'rc': 1,
+        }
+
+    if not os.path.isfile(signature):
+        message = (
+            "cannot verify signature of '{file}', signature '{signature}' does not exist"
+        ).format(file=file, signature=signature)
+        logger.debug(message)
+        return {
+            'gpg': False,
+            'stdout': '',
+            'stderr': message,
+            'rc': 1,
+        }
+
+    keys = [constants.pub_gpg_path]
+    _extra_key = os.getenv("EXTRA_GPG_KEY")
+    if _extra_key:
+        keys.append(_extra_key)
+
+    for key in keys:
+        if not os.path.isfile(key):
+            message = (
+                "cannot verify signature of '{file}', key '{key}' does not exist"
+            ).format(file=file, key=key)
+            logger.debug(message)
+            return {
+                'gpg': False,
+                'stdout': '',
+                'stderr': message,
+                'rc': 1,
+            }
+
+    home = tempfile.mkdtemp()
+
+    logger.debug("importing the public keys into temporary environment")
+    for key in keys:
+        stdout, stderr, rc = _call_gpg(home, "--import {key}".format(key=key))
```

Comment:
To go along with the change to how command is parsed in the `_call_gpg()` function:
```suggestion
        stdout, stderr, rc = _call_gpg(home, ["--import",key])
```